#!/usr/bin/env python
PACKAGE = "pfvtr"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

size_enum = gen.enum([ gen.const("SIFT", int_t, 0, "SIFT"),
                  gen.const("SURF", int_t, 1, "SURF"),
                  gen.const("KAZE", int_t, 2, "KAZE"),
                  gen.const("AKAZE", int_t, 3, "AKAZE"),
                  gen.const("BRISK", int_t, 4, "BRISK"),
                  gen.const("ORB", int_t, 5, "ORB")],
                  "An enum to set feature type")

gen.add("feature_type", int_t, 0, "Which feature to use", 0, 0, 5, edit_method=size_enum)

exit(gen.generate(PACKAGE, "pfvtr", "LiveFeatures"))
